src/AppInstallerCommonCore/Public/winget/ManifestCommon.h

```diff
@@ -135,16 +135,36 @@ namespace AppInstaller::Manifest
     {
         DependencyList() = default;
 
-        void Add(const Dependency& dependency) 
+        void Add(const Dependency& newDependency)
         { 
-            dependencies.push_back(dependency); 
+            Dependency* existingDependency = this->HasDependency(newDependency);
+
+            if (existingDependency != NULL) {
+                if (newDependency.MinVersion) 
+                {
+                    if (existingDependency->MinVersion)
+                    {
+                        const auto& newDependencyVersion = AppInstaller::Utility::Version(newDependency.MinVersion.value());

```

Comment:
I think it would also be fine to simply hold the MinVersion as `std::optional<Version>`; but that can be an optimization to think about in a future change.

Reply:
Agree, as of now that is the only place we are in need of a Version object but it will surely change in the future.